core/sdk-core/src/test/java/utils/SdkSubscriberTest.java

```diff
@@ -146,10 +146,10 @@ public void limiting_subscriber_with_multiple_thread_publishers() throws Interru
         final int limitFactor = 5;
         LimitingSubscriber<Integer> limitingSubscriber = new LimitingSubscriber<>(mockSubscriber, limitFactor);
         limitingSubscriber.onSubscribe(new EmptySubscription(mockSubscriber));
-        final ExecutorService executorService = Executors.newFixedThreadPool(10);
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
         for (int i = 0; i < 10; i++) {
-            final Integer integer = Integer.valueOf(i);
-            executorService.submit(() -> limitingSubscriber.onNext(new Integer(integer)));
+            int integer = i;
+            executorService.submit(() -> limitingSubscriber.onNext(integer));
```

Comment:
interesting, does the call to onNext actually autobox the int after the change? 

Otherwise if we want to submit an Integer we can remove the extra call and do 

```
Integer boxed = Integer.valueOf(i);
executorService.submit(() -> limitingSubscriber.onNext(boxed));
```

Reply:
Tests are passing, but updated to
`Integer boxed = i;`